ports/fastrtps/portfile.cmake

```
@@ -1,30 +1,28 @@
-# https://github.com/eProsima/Fast-DDS/pull/3983
-# Could remove after 2.6.6/2.10.2/2.11.2, other minor versions not applied
-# Could remove after 2.13 released, or any newer mijor versions
-vcpkg_download_distfile(
-    PR_3983_PATCH
-    URLS https://github.com/eProsima/Fast-DDS/commit/2601e95429f5676c47c25200e995fb93157e7815.patch?full_index=1
-    SHA512 440ee6a918d7085b6520613fad1482a9b833ee259e64c8919bdeb43277f3685362e6314380bafc6c51dad46fd16a9b415343e9db28f157d13f76b9af0cb21e8d
-    FILENAME fastrtps-2601e95429f5676c47c25200e995fb93157e7815.patch
-)
 
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO eProsima/Fast-DDS
-    REF v2.7.0
-    SHA512 289c94fb177209ffc80e93ae61822c83e7cb74ba7682f05a921c50ce048498bd811c19825d1fdb8af39b29a64904e96d87c5c59468139f0d8bb528549b80c94a
+    REF v2.14.0
+    SHA512 cf3628ac7598c83f09e3975de9fb59e868797adc331d7b4a226c38c0bbd1f7ce15deeb7cedee938a86a68426585e0b5646a84f40705c9a04553b7e37f20b2a4d
     HEAD_REF master
     PATCHES
         fix-find-package-asio.patch
         disable-symlink.patch
-        fix-xtime.patch
-        fix_thread.patch #https://github.com/eProsima/Fast-DDS/pull/3904
-        "${PR_3983_PATCH}"
 )
 
+set(extra_opts "")
+if (VCPKG_TARGET_IS_WINDOWS AND VCPKG_TARGET_ARCHITECTURE STREQUAL "arm64")
+    # when cross-compiling, try_run will not work.
+    set(extra_opts
+        -DSM_RUN_RESULT=TRUE
+        -DSM_RUN_RESULT__TRYRUN_OUTPUT=''

```

Is the output really expected to contain single quotes?
If the output is expected to be empty:
```suggestion
        -DSM_RUN_RESULT__TRYRUN_OUTPUT=
```

---

... usually it is enough to set `try_run`'s output variable, not intermediate variables logged by CMake.

---

Thanks for the advice. I'm not sure if this is necessary, I just took this from another port and adapted it. As I cannot test this locally, I will remove this line and let the CI figure out if it works.

---

Nope, if I do not specify `SM_RUN_RESULT__TRYRUN_OUTPUT` cmake complains:
```
CMake Error: try_run() invoked in cross-compiling mode, please set the following cache variables appropriately:
   SM_RUN_RESULT (advanced)
   SM_RUN_RESULT__TRYRUN_OUTPUT (advanced)
```

---

> I just took this from another port and adapted it.

Oh. Is this even for the same question, shared mutex?
https://github.com/eProsima/Fast-DDS/blob/5d1ecac569bb239ee616290a67473bed20ac4ca3/src/cpp/CMakeLists.txt#L371-L387
(Does the port sufficiently de-vendor deps? :see_no_evil:)

---

Yes, this treatment is for the shared mutex stuff. 

> Does the port sufficiently de-vendor deps? 

I don't get what that means, but any recommendations on how to improve this are welcome.

---

> Yes, this treatment is for the shared mutex stuff.

I mean: Did you take it from a port where it also solves the shared mutex capability detection?

> > Does the port sufficiently de-vendor deps? 

I was wondering if the package carries (copies of) third-party dependencies which are not properly turned into pure vcpkg dependencies. This PR doesn't need to improve this for existing dependencies. I just stumbled over it because boost is mentioned in the context of the fallback implementation of shared mutex support.